server/src/test/java/umm3601/user/UserControllerSpec.java

```diff
@@ -480,13 +484,14 @@ void addInvalidEmailUser() throws IOException {
 
   @Test
   void addInvalidAgeUser() throws IOException {
-    String testNewUser = "{"
-        + "\"name\": \"Test User\","
-        + "\"age\": \"notanumber\","
-        + "\"company\": \"testers\","
-        + "\"email\": \"test@example.com\","
-        + "\"role\": \"viewer\""
-        + "}";
+    String testNewUser = """
+        {
+          "name": "Test User",
+          "age": "notanumber",
+          "company": "testers",
+          "email": test@example.com",
```

Comment:
I think a quote is missing before the given email here

Reply:
You are correct! I'll fix that.

Reply:
Thanks to @kklamberty for noticing that I was missing an open quote on the `"email"` field in the `addInvalidAgeUser()` test.

Weirdly, this totally passed the test because it _also_ generated a `ValidationExcption`. In doing some looking around, I couldn't find any easy way to figure out _where_ validation failed, so that we could distinguish between an intentional error like `"age": "notanumber"` and a type error like leaving off a quote.

That's a bummer because it's pretty easy to have a typo in this kind of "uninterpreted" JSON, and this means we would never know unless some eagle-eye like KK notices.

If someone knows of a better way to check for this, definitely let us know!